client/layout/index.jsx

```diff
@@ -295,6 +296,7 @@ class Layout extends Component {
 				<QuerySites primaryAndRecent={ ! config.isEnabled( 'jetpack-cloud' ) } />
 				{ this.props.shouldQueryAllSites && <QuerySites allSites /> }
 				<QueryPreferences />
+				<QuerySiteFeatures siteId={ this.props.siteId } />
```

Comment:
I'm not sure where it's triggered (perhaps as part of QuerySites) but the features information is being returned under the `plan` section of the data that comes back from the `/sites/<SITE ID>` API call. Could we save a request by using that data and only explicitly querying for it if it's unavailable, or we suspect it's stale?

Reply:
I decided to try this out by checking for the features under the site plan data in the selector, and then checking that selector before rendering the `<QuerySiteFeatures>` component. If we already have the features from either /sites or /features, then we shouldn't fetch the features. I also prevented fetching if we were already fetching /site or /sites.

What kept happening was that state would update with the features from /sites, but then update without them afterwards. And then of course we'd have to fetch the features again.

The cause seems to be that while the `v1.1/sites/<SITE ID>` API response does include features in the plan section, the `v1.2/me/sites` API response we fetch afterwards on line 297 (`<QuerySites allSites />`) does not include features. That replaces the site data so that features are removed from state. 

Should we update the `v1.2/me/sites` endpoint to include features? Or is there another way of working around this that I missed?

Reply:
Also, for some reason, the `plan` section is not populated in the endpoint response. I tested with retrosimplepremium01.wordpress.com site.